kube-runtime/src/cache/mod.rs

```diff
@@ -25,6 +29,87 @@ where
     stream.inspect_ok(move |event| store.apply_watcher_event(event))
 }
 
+/// A simple reflector cache around a store and an owned watcher
+pub struct Cache<K>
```

Comment:
I don't know, I know we have an education issue, but this feels like it singles out the reflector/cache layer too much for my taste.

Reply:
To me `Cache` sounds like a synonym of `Store`, not what's effectively a builder for it.

Reply:
That said, I'm :+1: on renaming the module to `cache` (and maybe renaming `Store` to `Cache`?).

Reply:
I'm not attached to names here. Happy to let `Store` take that name, that seems reasonable to me.

>  but this feels like it singles out the reflector/cache layer too much for my taste

Yeah, it's not my intention to just single out the `reflector` here. I was also hoping to do something to `watcher` for the same error issue you raised + the ergonomics issue, but I thought I'd start with this as a place to start and see what you thought.

We could potentially wrap both with one struct:

- if users call `Cache::store()` we could record events
- if the users does not call `Cache::store()` we could simply forward a watcher

Reply:
Have renamed it to `Reflector` for now, but going to park this for a little bit. The `watcher` + `backoff` integration needs to be attempted before this so that we can figure out how to best expose that through something higher level. If the `watcher` gets a third parameter for this, then even more a case for a helper wrappers on top of it. Will continue after attempting that.